database/migrations/2025_07_30_000000_create_api_logs_table.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::create('api_logs', function (Blueprint $table) {
+            $table->id();
+            $table->string('api_version', 10); // 'internal', 'v1', 'v2'.
+            $table->unsignedBigInteger('user_id')->nullable();
+            $table->string('endpoint', 255);
+            $table->string('method', 10);
+            $table->integer('response_code');
+            $table->integer('response_time_ms')->nullable();
+            $table->string('ip_address', 45);
+            $table->text('user_agent')->nullable();
+            $table->json('request_data')->nullable();
+            $table->string('error_message')->nullable();
+            $table->timestamp('created_at')->useCurrent();
```

Comment:
Would it be worth also capturing the response length?

Reply:
Yeah, I think so. Updated the migration & logging stuff in latest.

<img width="1059" height="134" alt="Screenshot 2025-07-31 at 8 14 27 PM" src="https://github.com/user-attachments/assets/8270074c-6733-46ed-86a3-05243deffd57" />

Reply:
Minor: `response_size_bytes` is NULL for laravel errors. You can see it the second row of your image. The 403 error from using a valid token for an unauthorized user is captured properly.

Reply:
This should now be resolved in latest:
<img width="716" height="81" alt="Screenshot 2025-08-14 at 8 07 37 PM" src="https://github.com/user-attachments/assets/61c7bdad-2732-4b72-8686-47c05c91e788" />